script.lua

```diff
@@ -5729,6 +5729,18 @@ function setupPlayerArea(params)
     function Elements:__tostring()
         return table.concat(self, "")
     end
+    local function modifyElements(pars)
```

Comment:
for syntax purposes, i've typically been calling the parameter `params`

Reply:
I did that initially, but on first push, the linter complained that the `params` argument in in `modifyElements()` was shadowing the `params` argument in the outer `setupPlayerArea()` function. `modifyElements()` has to be called inside `setupPlayerArea()`, 'cause the definition of `Elements` sits inside `setupPlayerArea()`.

I'm not actually sure why `Elements` is defined inside `setupPlayerArea()`. If this definition could be moved outside `setupPlayerArea()`, then the definition of `modifyElements()` could likewise be moved outside, thus dodging any shadowing problems. I wonder if it would then also be possible to access the definition of `Elements` from the `aidBoard` script, 'cause I noticed that the `aidBoard` script contains a lot of lines of duplicated code to define `Elements` again.

So, do you know why `Elements` is defined inside `setupPlayerArea()`? Or a better way of avoiding the linter error?

Reply:
i dont remember why elements is defined in there, Tom (the guy who wrote the python plugin to deconstruct and construct tts save files) was the one who wrote this. i imagine it can be moved outside

Reply:
I'll move it outside. Given it's duplicated in `aidBoard`, I might do something about trying to avoid that duplication, as well. I think I need to read up on metatables to know how to do that properly. I did notice that one of Tom's old pull requests (#48) mentioned that `tts-tools` can handle Lua modules, so I may also look into that as a possibility.

Reply:
Ah, looks like Tom pulled support for modules in `tts-tools` (tomprince/tabletoptools#21), so I won't be using a module for this.

Reply:
is it not possible to move `Elements` to live outside of the current scope it's in while still having it exist in this file?

Reply:
Oh, it's certainly possible, and I'm treating it as a backup option. But the definition of `Elements` is over 30 lines of code that's duplicated four times (in Global, `aidBoard`, `PowerEditor` and `SpiritEditor`, so it'd be really nice to get those consolidated. I'm still investigating whether that's possible.

Reply:
Alas, I can't find any way this is possible. I've just moved the global copy of `Elements` outside of `setupPlayerArea()`, and not touched the other three copies.